msteams-platform/bots/how-to/conversations/send-proactive-messages.md

```diff
@@ -7,6 +7,153 @@ ms.author: anclear
+- To handle notifications more smoothly, consider other ways to integrate the notification into the conversation flow, such as setting a flag in the conversation state or adding the notification to a queue.
+
+Proactive messages generally fall into one of the categories described below.
+
+### Welcome messages
+
+When receiving a welcome message a users do not have context. Also this is the first time they have interacted with the bot which is an opportunity to create a good first impression. Well structured messages include:
+
+- **Why the users are receiving the message.** It should be very clear to the users why they are receiving the message. If your bot was installed in a channel and you sent a welcome message to all users, let them know what channel it was installed and potentially who installed it.
+- **What is the offer.** What can they do with your bot? What value it brings?
+- **What should they do next.** Invite them to try out a command, or interact with your bot in some way.
+
+### Notification messages
+
+When sending notifications, assure that the users have a clear path to take common actions, and a clear understanding of why the notification ocurred. Well structured messages include:
+
+- **What happened.** A clear indication of what caused the notification.
+- **What it happened to.** What item was updated to cause the notification.
+- **Who did it.** Who took the action that caused the notification to be sent.
+- **What they can do about it.** Make it easy for the users to take actions based on the notification.
+- **How they can opt out.** You need to provide a path for users to opt out of additional notifications.
+
+## Obtain necessary user information
+
+Bots can create new conversations with an individual Microsoft Teams user by obtaining the userâ€™s *unique ID* and *tenant ID.* You can obtain these values using one of the following methods:
+
+> [!WARNING]
+> Please, verify these links; they point to topics in the _old folder.
+
+- By [fetching the team roster](../../../_old/concepts/bots/bots-context.md#fetching-the-team-roster) from a channel your app is installed in.
+- By caching them when a user [interacts with your bot in a channel](../../../_old/concepts/bots/bot-conversations/bots-conv-channel.md).
+- When a users is [@mentioned in a channel conversation](../../../_old/concepts/bots/bot-conversations/bots-conv-channel.md#-mentions) the bot is a part of.
+- By caching them when you [receive the conversationUpdate](../../../_old/concepts/bots/bots-notifications.md#team-member-or-bot-addition) event when your app is installed in a personal scope, or new members are added to a channel or group chat that
+
+### Proactively install your app using Graph
```

Comment:
We don't [in general] document preview and beta features in the main documentation. Check with Andrew whether the information in this section can be put in a separate doc/location.